src/Stryker.Core/Stryker.Core/Compiling/RollbackProcess.cs

```diff
@@ -134,30 +134,30 @@ private void ScanAllMutationsIfsAndIds(SyntaxNode node,  IDictionary<SyntaxNode,
 
         private void DumpBuildErrors(KeyValuePair<SyntaxTree, ICollection<Diagnostic>> syntaxTreeMap)
         {
-            _logger.LogInformation($"Roll backing mutations from {syntaxTreeMap.Key.FilePath}.");
+            Logger.LogInformation($"Roll backing mutations from {syntaxTreeMap.Key.FilePath}.");
             var sourceLines = syntaxTreeMap.Key.ToString().Split("\n");
             foreach (var diagnostic in syntaxTreeMap.Value)
             {
                 var fileLinePositionSpan = diagnostic.Location.GetMappedLineSpan();
-                _logger.LogInformation($"Error :{diagnostic.GetMessage()}, {fileLinePositionSpan.ToString()}");
+                Logger.LogInformation($"Error :{diagnostic.GetMessage()}, {fileLinePositionSpan.ToString()}");
                 for (var i = Math.Max(0, fileLinePositionSpan.StartLinePosition.Line - 1);
                     i <= Math.Min(fileLinePositionSpan.EndLinePosition.Line + 1, sourceLines.Length - 1);
                     i++)
                 {
-                    _logger.LogInformation($"{i + 1}: {sourceLines[i]}");
+                    Logger.LogInformation($"{i + 1}: {sourceLines[i]}");
                 }
             }
 
-            _logger.LogInformation(Environment.NewLine);
+            Logger.LogInformation(Environment.NewLine);
         }
 
-        private SyntaxTree RemoveMutantIfStatements(SyntaxTree originalTree, IEnumerable<Diagnostic> diagnosticInfo,
-            bool lastAttempt, bool devMode)
+        private SyntaxTree RemoveMutantIfStatements(SyntaxTree originalTree, IEnumerable<Diagnostic> diagnosticInfo, bool devMode)
         {
             var rollbackRoot = originalTree.GetRoot();
             // find all if statements to remove
             var brokenMutations = new Collection<SyntaxNode>();
-            foreach (var diagnostic in diagnosticInfo)
+            var diagnostics = diagnosticInfo as Diagnostic[] ?? diagnosticInfo.ToArray();
```

Comment:
Why are you casting to array here?

Reply:
performance reason, as diagnostics is reused.